### ISSUE TITLE
[HOLD for payment 2023-07-10] Hide threads from LHN if they don't have a comment yet

### ISSUE THREAD

**Issue by grgia** [Expensify/App#19205]
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01aee66cc771604fe5</li>
        <li>Upwork Job ID: 1659267104086478848</li>
        <li>Last Price Increase: 2023-05-18</li>
    </ul>
</details>

**Comment by grgia** [comment]
@chiragsalian some thoughts-

we can 1) simply filter out reports that have no visible actions on the parentReportAction aka this branch: https://github.com/Expensify/App/pull/new/georgia-hideEmptyThreads

or 2) we don't start adding users to the participants list until at least one comment has been made

**Comment by chiragsalian** [comment]
i would suggest 1, because if we go with 2 then that's changing the participant list data to work drastically different for threads. Plus with solution 2, then once the user is in the chat and they click members they won't see themselves which is incorrect. 
So option 1 sounds correct to me.

**Comment by grgia** [comment]
draft PR up, but need to fix the optimistic data

**Comment by grgia** [comment]
still in progress

**Comment by grgia** [comment]
I need to update my PR to fix the optimistic data

**Comment by JmillsExpensify** [comment]
Isn't this a duplicate of this [older issue](https://github.com/Expensify/App/issues/14523)?

**Comment by PauloGasparSv** [comment]
> Isn't this a duplicate of this https://github.com/Expensify/App/issues/14523?

Hey @JmillsExpensify since this issue is specific to Threads I don't think it is a duplicate! I checked the draft P.R. logic and compared it with [the one you mentioned](https://github.com/Expensify/App/pull/19321) and since I'm adding another clause and param to `shouldReportBeInOptionList` I don't think we have any conflicts in logic either : )

**Comment by grgia** [comment]
Still need to update unit tests

**Comment by grgia** [comment]
Coming back to this today

**Comment by grgia** [comment]
@MitchExpensify added BZ label to help with C+ $1000 PR Review Payment for this PR - https://github.com/Expensify/App/pull/19375

**Comment by MitchExpensify** [comment]
Upwork job created for C+ payment and hire offer sent to @mollfpr https://www.upwork.com/jobs/~01393c0b271fdfe3bc

**Comment by MitchExpensify** [comment]
Added a reminder to pay on the 10th. Friendly reminder to take a look at the BZ checklist above @mollfpr 👍

**Comment by Julesssss** [comment]
Hey, just curious how this should work for tasks? Currently uncommented [tasks disappear ](https://github.com/Expensify/App/issues/22300)which doesn't seem expected

**Comment by MitchExpensify** [comment]
Paid and contract ended! @mollfpr

**Comment by MitchExpensify** [comment]
Are their any tests we should add for this change? https://github.com/Expensify/App/issues/19205#issuecomment-1618405750

**Comment by mollfpr** [comment]
We need the regression step here.

**_Web/Desktop_/Large screen**
1. Create a thread from a message
2. **Verify** that the thread report appears in the LHN
3. Begin Typing a message; note that the thread still appears in LHN with the edit icon.
4. Click another report
5. **Verify** that the threaded message with no replies no longer appears in the LHN
6. Create a thread from a message, and comment
7. Click another report
8. **Verify** that the threaded message with no replies remains in the LHN

**mWeb/Mobile/Small screen**

1. Create a thread, don't comment
2. Go back to LHN/chat list page
3. **Verify** that the newly created thread does not appear

**Comment by MitchExpensify** [comment]
Issue to add new test [here](https://github.com/Expensify/Expensify/issues/298839)
